libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -57,27 +59,47 @@ internal class MapboxTripSession(
     companion object {
         @Volatile
         internal var UNCONDITIONAL_STATUS_POLLING_PATIENCE = 2000L
+
         @Volatile
         internal var UNCONDITIONAL_STATUS_POLLING_INTERVAL = 1000L
+
+        private const val TAG = "MapboxTripSession"
     }
 
     private var updateNavigatorStatusDataJobs: MutableList<Job> = CopyOnWriteArrayList()
 
     override var route: DirectionsRoute? = null
         set(value) {
+            val isSameRoute = field.isSameRoute(value)
             field = value
             if (value == null) {
                 routeAlerts = emptyList()
                 routeProgress = null
             }
             cancelOngoingUpdateNavigatorStatusDataJobs()
-            val setRouteJob = threadController.getMainScopeAndRootJob().scope.launch {
-                navigator.setRoute(value)?.let {
-                    routeAlerts = it.routeAlerts
+            val updateRouteJob = threadController.getMainScopeAndRootJob().scope.launch {
+                if (isSameRoute) {
+                    value?.legs()?.forEachIndexed { index, routeLeg ->
+                        routeLeg.annotation()?.toJson()?.let { annotations ->
+                            navigator.updateAnnotations(annotations, index).let { success ->
```

Comment:
Do we want to run this in the `Main` thread? Noting that `navigator.setRoute` switches to `NavigatorDispatcher` but `navigator.updateAnnotations` doesn't. Maybe a regression?

Reply:
Right, `updateAnnotations` should switch to `NavigatorDispatcher` as well.